src/aiohomeconnect/client.py

```diff
@@ -676,21 +703,36 @@ async def get_all_events(
         * [Program progress changes](https://api-docs.home-connect.com/events#program-progress-changes)
         * [Home appliance state changes](https://api-docs.home-connect.com/events#home-appliance-state-changes)
         """
-        response = await self._auth.request(
+        # We use 60 seconds timeout because at least every 55 seconds a KEEP-ALIVE event
+        # will be sent. See https://api-docs.home-connect.com/events/#availability-matrix
+        async with self._auth.connect_sse(
             "GET",
             "/homeappliances/events",
-            headers={"Accept-Language": accept_language},
-        )
-        return ArrayOfEvents.from_dict(response.json()["data"])
-
-    async def get_events(
+            timeout=Timeout(60),
+            headers={
+                "Accept-Language": accept_language,
+            },
+        ) as event_source:
+            async for sse in event_source.aiter_sse():
+                if sse.event == EventType.KEEP_ALIVE:
+                    continue
+                data = json.loads(sse.data)
+                if "items" in data:
+                    for item in data["items"]:
+                        item["haId"] = data["haId"]
```

Comment:
Side note: It would be nice to be able to move this part of the code to the model package since it pertains to the API model. We can look at that later though. For now, it's ok.